runtime/common/src/slots.rs

```diff
@@ -421,6 +421,43 @@ impl<T: Config> Leaser for Module<T> {
 	fn lease_period_index() -> Self::LeasePeriod {
 		<frame_system::Pallet<T>>::block_number() / T::LeasePeriod::get()
 	}
+
+	fn already_leased(
+		para_id: ParaId,
+		first_period: Self::LeasePeriod,
+		last_period: Self::LeasePeriod,
+	) -> bool {
+		let current_lease_period = Self::lease_period_index();
+
+		// Can't look in the past, so we pick whichever is the biggest.
+		let start_period = first_period.max(current_lease_period);
+		// Find the offset to look into the lease period list.
+		// Subtraction is safe because of max above.
+		let offset = match (start_period - current_lease_period).checked_into::<usize>() {
+			Some(offset) => offset,
+			None => return false,
```

Comment:
to be pedantic you could consider making these an `Err` rather than false, but might be an overkill. If the call site gave a `first_period` so large not to fit into usize then they are clearly wrogn anyhow.

Reply:
I think `return true` would be the safest choice in the unlikely scenario that the conversion fails, worst case you prevent some crowdloan from happening.

Reply:
```suggestion
			None => return true,
```